From: Justus Winter <justus@g10code.com>
Date: Tue, 28 Feb 2017 13:15:42 +0100
Subject: gpg,tools: Make trust-model configurable via gpgconf.

* g10/gpg.c (gpgconf_list): Add 'trust-model'.
* tools/gpgconf-comp.c (gc_options_gpg): Likewise.

GnuPG-bug-id: 2381
Signed-off-by: Justus Winter <justus@g10code.com>
(cherry picked from commit ebeccd73eb85f9027f0985d77dfe901266c6ddef)
---
 g10/gpg.c            | 1 +
 tools/gpgconf-comp.c | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/g10/gpg.c b/g10/gpg.c
index 0c5a1677c..09bdf66ba 100644
--- a/g10/gpg.c
+++ b/g10/gpg.c
@@ -1845,6 +1845,7 @@ gpgconf_list (const char *configfile)
   es_printf ("group:%lu:\n", GC_OPT_FLAG_NONE);
   es_printf ("compliance:%lu:\"%s:\n", GC_OPT_FLAG_DEFAULT, "gnupg");
   es_printf ("default-new-key-algo:%lu:\n", GC_OPT_FLAG_NONE);
+  es_printf ("trust-model:%lu:\n", GC_OPT_FLAG_NONE);
 
   /* The next one is an info only item and should match the macros at
      the top of keygen.c  */
diff --git a/tools/gpgconf-comp.c b/tools/gpgconf-comp.c
index a0d965969..cdd2586b7 100644
--- a/tools/gpgconf-comp.c
+++ b/tools/gpgconf-comp.c
@@ -716,6 +716,10 @@ static gc_option_t gc_options_gpg[] =
      (GC_OPT_FLAG_ARG_OPT|GC_OPT_FLAG_NO_CHANGE), GC_LEVEL_INVISIBLE,
      NULL, NULL,
      GC_ARG_TYPE_STRING, GC_BACKEND_GPG },
+   { "trust-model",
+     GC_OPT_FLAG_NONE, GC_LEVEL_INVISIBLE,
+     NULL, NULL,
+     GC_ARG_TYPE_STRING, GC_BACKEND_GPG },
 
 
    { "Debug",
